src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -95,4 +127,24 @@ protected synchronized void write() throws IOException {
     // atomic rotate
     FileRotate.rotateFile(tmpPath, pathToFile);
   }
+
+  /**
+   * Attempts to write the window data to disk if enough time has passed since the last write
+   *
+   * <p>If this method tries and fails to write data to disk, it will print an error message but will
+   * not throw any exception, responsibility is on the caller
```

Comment:
Since *responsibility for error handling is on the caller*, we should just throw IOException here and let the caller log it or handle it as needed.

Reply:
Done